### PR TITLE
server: set `system_time_zone` from `system_tz`

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve? 
This PR tries to close https://github.com/pingcap/tidb/issues/13894, by setting system variable `system_time_zone` from `system_tz` variable in `mysql.tidb` table.

### What is changed and how it works?
A `setSystemTimeZoneVariable()` is added to `server.NewServer()`, which get the value of `systemTZ` from exported function `GetSystemTZ()`

Please notice that this PR doesn't change any other behavior: all time-related calculations are not affected.

### Check List 

Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)
 
```
export TZ="Europe/London"
# Before the initial bootstrap of TiDB Server)

./bin/tidb-server
# TiDB bootstrapped and infer `systemTZ` from TZ environment variable.

tidb> select variable_value from mysql.tidb where variable_name = "system_tz"; 
+----------------+
| VARIABLE_VALUE |
+----------------+
| Europe/London  |
+----------------+
1 row in set (0.00 sec)

tidb> select @@system_time_zone;                                                  
+--------------------+
| @@system_time_zone |
+--------------------+
| Europe/London      |
+--------------------+
1 row in set (0.00 sec)
```

The scripts described in #13894 is also used for manual tests.

Code changes

 - Has exported function/method change
 
Side effects

 - NA

Related changes

 - Need to update the documentation

Release note

 - NA

### CODE REVIEW

**Comment by bb7133** [comment]
PTAL @lysu @zhexuany

**Comment by Deardrops** [comment]
LGTM

File: server/tidb_test.go [hunk 1]
```diff
@@ -293,6 +294,19 @@ func registerTLSConfig(configName string, caCertPath string, clientCertPath stri
...
+	tk := testkit.NewTestKit(c, ts.store)
+	cfg := config.NewConfig()
+	cfg.Port = 4002
+	cfg.Status.ReportStatus = false
+	server, err := NewServer(cfg, ts.tidbdrv)
+	c.Assert(err, IsNil)
+	defer server.Close()
+
+	tz1 := tk.MustQuery("select variable_value from mysql.tidb where variable_name = 'system_tz'").Rows()
+	tk.MustQuery("select @@system_time_zone").Check(tz1)
```
Comment (AilinKid) [inline server/tidb_test.go]: Need test check equal to  TZ from host?

**Pull request merged**